File: 3.go
package main

import "fmt"

// Функция для реверса цифр числа
func reverseNumber(num int) int {
    reversed := 0
    for n := num; n > 0; n /= 10 {
        reversed = reversed*10 + n%10
    }
    return reversed
}

func main() {
    var n int
    fmt.Print("Введите количество чисел: ")
    fmt.Scan(&n)

    fmt.Print("Введите числа: ")
    for i := 0; i < n; i++ {
        var num int
        fmt.Scan(&num) // Чтение каждого числа по отдельности

        reversed := reverseNumber(num)
        difference := num - reversed
        if difference < 0 {
            difference = -difference
        }

        fmt.Printf("|%d - %d| = %d\n", num, reversed, difference)
    }
}

File: 3.rs
use std::io;

// Функция для реверса цифр числа
fn reverse_number(num: i32) -> i32 {
    let mut reversed = 0;
    let mut n = num;
    while n > 0 {
        reversed = reversed * 10 + n % 10;
        n /= 10;
    }
    reversed
}

fn main() {
    // Чтение количества чисел
    println!("Введите количество чисел:");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Ошибка чтения");
    let n: usize = input.trim().parse().expect("Ожидается число");

    // Чтение чисел
    println!("Введите числа через пробел:");
    input.clear();
    io::stdin().read_line(&mut input).expect("Ошибка чтения");
    
    // Обработка и вывод результатов
    println!("Результаты:");
    input
        .split_whitespace()
        .take(n)
        .map(|s| s.parse::<i32>().expect("Ожидается число"))
        .for_each(|num| {
            let reversed = reverse_number(num);
            let difference = (num - reversed).abs();
            println!("|{} - {}| = {}", num, reversed, difference);
        });
}
